.github/workflows/tests-kmip.yaml

```
@@ -0,0 +1,49 @@
+---
+name: tests-kmip
+
+on:
+  push:
+    branches-ignore:
+    - 'development/**'
+    paths:
+    - .github/workflows/tests-kmip.yaml
+    - .github/docker-compose.yaml
+    - .github/pykmip/**
+    - lib/network/kmip/**
+    - tests/functional/kmip/**
```

`paths` does not work as you would expect with `push` when working on the feature branches, and does not work well with gitwaterflow integration branches as well (`w/**`).

As a quick fix I can recommend you to use a combination of `push` and `pull_request` trigger, even though it won't be perfect, it will work better:

```suggestion
  pull_request:
    branches:
    - development/*
    - hotfix/*
    paths:
    - .github/workflows/tests-kmip.yaml
    - .github/docker-compose.yaml
    - .github/pykmip/**
    - lib/network/kmip/**
    - tests/functional/kmip/**
  push:
    branches:
    - q/*
    - w/**
    paths:
    - .github/workflows/tests-kmip.yaml
    - .github/docker-compose.yaml
    - .github/pykmip/**
    - lib/network/kmip/**
    - tests/functional/kmip/**
```

For a more accurate way I recommend you checking out the use of https://github.com/tj-actions/changed-files/, we do it in Artesca for the documentation build, if you want to see a real life example of it. 

But honestly, if it doesn't take too long, I would just leave it. I see no issue in having it around:

```suggestion
  push:
    branches-ignore:
    - 'development/**'
```

---

let's try with both push and pull_request